lib/synapse/service_watcher/multi/resolver/sequential.rb

```diff
@@ -0,0 +1,84 @@
+require 'synapse/service_watcher/multi/resolver/base'
+require 'synapse/log'
+require 'synapse/statsd'
+require 'synapse/atomic'
+
+class Synapse::ServiceWatcher::Resolver
+  class SequentialResolver < BaseResolver
+    include Synapse::Logging
+    include Synapse::StatsD
+
+    def initialize(opts, watchers, reconfigure_callback)
+      @watcher_order = opts['sequential_order']
+      super(opts, watchers, reconfigure_callback)
+
+      @watcher_setting = Synapse::AtomicValue.new(@watcher_order[0])
+    end
+
+    def start
+      log.info "synapse: sequential resolver: starting"
+      pick_watcher
+    end
+
+    def validate_opts
+      raise ArgumentError, "sequential resolver expects method to be union" unless @opts['method'] == 'sequential'
```

Comment:
typo?  "expects method to be union" - > "expects method to be sequential".
Might be more useful error message if you include what it detected as the `@opts['method']` value as well